File: JS/Basics/4_3_palindromeNumber.js
function reverseNumber(n) {
    let num = n;
    let reversedNum = 0;

    while (num > 0) {
        let rem = num % 10;
        reversedNum = reversedNum * 10 + rem;
        num = Math.floor(num / 10);
    }

    return reversedNum;
}

function palindromeNumber(n) {
    let reversedNumber = reverseNumber(n);

    if (reversedNumber == n) return true;
    else return false;
}

function main() {
    let num = 567;

    if (palindromeNumber(num)) console.log("Palindrome");
    else console.log("Not palindrome");
}

main();

File: C/Basics/4_3_palindromeNumber.c
// Check if the number is palindrome
// approach -> find reverse and then compare both

#include <stdio.h>

int main()
{
    int N = 123321;
    int num = N;
    int reverseNum = 0;
    int rem, isPalindrome = 1;

    while (num > 0)
    {
        rem = num % 10;
        num = num / 10;
        reverseNum = reverseNum * (10) + rem;
    }

       if (N == reverseNum)
    {
        printf("Entered num is palindrome : %d\n", N);
    }
    else
    {
        printf("Entered num is not palindrome : %d\n", N);
    }
}